integration-tests/tests/src/realm-constructor.js

```diff
@@ -0,0 +1,277 @@
+        it("throws when called with an empty string", () => {
+            expect(() => {
+                new Realm("");
+            }).to.throw(); // The actual message varies across environments
+        });
+
+        it("throws when called with two strings", () => {
+            expect(() => {
+                new Realm("", "");
+            }).to.throw("Invalid arguments when constructing 'Realm'");
+        });
+    });
+
+    describe("opening, closing and re-opening", () => {
+        it("changes the isClosed boolean", () => {
+            // Open the Realm
+            const realm = new Realm({ schema: [] });
+            expect(realm.isClosed).to.equal(false);
+            // Close it
+            realm.close();
+            expect(realm.isClosed).to.equal(true);
+            // Re-open it
+            const reopenedRealm = new Realm({ schema: [] });
+            expect(reopenedRealm.isClosed).to.equal(false);
+            // The first Realm instance is still considered closed
+            expect(realm.isClosed).to.equal(true);
+        });
+    });
+
+    describe("schema version", () => {
+        it("initiates with schema version 0", () => {
+            const realm = new Realm({ schema: [] });
+            expect(realm.schemaVersion).to.equal(0);
+        });
+
+        it("can reopen with the same version", () => {
+            // Open it ...
+            const realm = new Realm({ schema: [], schemaVersion: 0 });
+            expect(realm.schemaVersion).to.equal(0);
+            // Re-open with a different version
+            const reopenedRealm = new Realm({ schema: [], schemaVersion: 0 });
+            expect(reopenedRealm.schemaVersion).to.equal(0);
+        });
+
+        it("can close and reopen with different versions", () => {
+            // Open it ...
+            const realm0 = new Realm({ schema: [], schemaVersion: 0 });
+            expect(realm0.schemaVersion).to.equal(0);
+            expect(realm0.schema).to.deep.equal([]);
+            realm0.close();
+            // Re-open with a different version
+            const realm1 = new Realm({ schema: [], schemaVersion: 1 });
+            expect(realm1.schemaVersion).to.equal(1);
+            expect(realm1.schema.length).to.equal(0);
+            realm1.close();
+            // Re-open with a different version
+            const realm2 = new Realm({ schema: PersonAndDogsSchema, schemaVersion: 2 });
+            expect(realm2.schemaVersion).to.equal(2);
+            expect(realm2.schema.length).to.equal(2);
+            // Add an object, using the schema
+            realm2.write(() => {
+                realm2.create("Person", { name: "John Doe", age: 42 });
+            });
+            // Expect an object
+            const persons = realm2.objects("Person");
+            expect(persons.length).to.equal(1);
+            expect(persons[0].name).to.equal("John Doe");
+            expect(persons[0].age).to.equal(42);
```

Comment:
`realm2.close()`

Reply:
I'm not doing anything after this that depends on the realm being closed and as far as I can tell `Realm.clearTestState` will close the Realm for me (it will eventually ask object store to clear all cached Realms, which will iterate and close them: https://github.com/realm/realm-object-store/blob/master/src/impl/realm_coordinator.cpp#L372)